src/linalg/cholesky.rs

```diff
@@ -200,6 +207,50 @@ where
         Some(Cholesky { chol: matrix })
     }
 
+    /// Attempts to compute the Cholesky decomposition of `matrix`, using the
+    /// given element in places where the decomposition would otherwise fail.
+    ///
+    /// The result can be garbage if your value is non-positive, resulting in
+    /// non-PSD matrix. Same caveats as with `new_unchecked` apply.
+    pub fn new_with_substitute_unchecked(mut matrix: OMatrix<T, D, D>, substitute: T) -> Self
```

Comment:
Having to copy the decomposition code for this is unfortunate. We should just modify the `Cholesky::new(matrix)` to this instead:
```rust
pub fn new(mut matrix: OMatrix<T, D, D>, substitute: Option<T>) -> Option<Self> {
```

Reply:
I'd say this would be unfortunate as in 99+% of the time, users don't need this functionality, so `new` should ideally just take a single matrix.

Perhaps instead we can implement `new` and the new function in terms of some internal function `new_with_substitute` that has the type signature that @sebcrozet proposed?

Reply:
> 
> 
> I'd say this would be unfortunate as in 99+% of the time, users don't need this functionality, so `new` should ideally just take a single matrix.
> 
> Perhaps instead we can implement `new` and the new function in terms of some internal function `new_with_substitute` that has the type signature that @sebcrozet proposed?

I think that would be unnecessarily complicated. Passing `Matrix::new(matrix, None)` doesn’t sound too noisy compared to having more construction methods.

Reply:
I'm sorry, but I completely disagree. If I see a function call `Cholesky::new(matrix)` it's clear what's going on. If I see `Cholesky::new(matrix, None)`, I have no idea what's going on and I have to look up the docs to see what the `None` is.

Reply:
Ok, that sounds legitimate, to make reading code easier. Let’s do it as you suggested then. We can add `pub fn new_with_substitute(matrix, substitute)` and then implement

```rust
pub fn new(matrix) -> Option<Self> { 
    Self::new_with_substitute(matrix, None) 
}
```

Reply:
I wanted to keep existing API as-is so that making a small release would be faster.

I did want to make `new` just call `new_with_substitute`, just forgot. Will do.